alerts/cluster-monitoring-operator/TargetDown.md

```diff
@@ -0,0 +1,269 @@
+# TargetDown
+
+## Meaning
+
+The alert means that Prometheus fails to scrape some targets.
+It fires when at least 10% of scrape targets in a Service are unreachable (`up`
+metric = 0) for at least 15 minutes.
+
+## Impact
+
+Metrics from the offline targets are not scraped by Prometheus. By consequence,
+we lose the visibility into the health and performance of the monitored application
+represented by the target.
+
+The alerts may lose accuracy or even become crippled if the missing metrics play
+a role in the triggering condition of the alert. Critical alerts may not fire
+when they should, potentially leading to unnoticed service disruption.
+
+Auto-scalers may not work properly if the missing metrics are critical to scaling
+conditions. This leads to potential degradation in user experience or waste of
+valuable resources.
+
+## Diagnosis & Mitigation
+
+The `TargetDown` alert indicates that one or more scrape targets are not responding
+to Prometheus scrape query.
+
+The alert has labels `namespace`, `service` and `job`. With these labels we can
+identify which Prometheus instance fails to scrape which target.
+
+In the Openshift web console, go to the tab **Observe** -> **Metrics**.  Check
+the metric `up` with these labels to find the down targets are in which pods.
+Here is an example query:
+```text
+up{namespace="openshift-monitoring", service="thanos-querier", job="thanos-querier"} == 0
+```
+
+Then check the label `prometheus` and `pod` in the returned metric `up`.
+The label `prometheus` should be either `openshift-monitoring/k8s` or `openshift-user-workload-monitoring/user-workload`,
+indicating the Prometheus pods scraping the target are either `prometheus-k8s-0`
+/ `prometheus-k8s-0` in the namespace `openshift-monitoring`, or `prometheus-user-workload-0`
+/ `prometheus-user-workload-1` in the namespace `openshift-user-workload-monitoring`.
+The label `pod` indicates the pod exposing the metrics endpoint.
+
+The targets are defined in either `ServiceMonitor` or `PodMonitor` resources.
+The labels `namespace` and `service` that help us locate the `Service` exposing
+the metric endpoint.
+The label `namespace` and `job` can locate the `ServiceMonitor` or `PodMonitor` that
+configures Prometheus to scrape the target. The `job` label is the name of the monitor.
+
+Now we have both end of the metric scraping flow as well as the monitor resources
+linking them together.
+We are ready to diagnosing the root cause by inspecting each component on the scraping
+workflow.
+
+
+### Network Issues
+
+Check the network connectivity between the Prometheus pod and the target pod.
```

Comment:
The `net_conntrack_dialer_conn_xxx_total` metrics exposed by Prometheus could be really helpful here

`net_conntrack_dialer_conn_failed_total` for example give the reason of conn failure: `timeout`. `resolution`, `refused` etc.

Reply:
I suggest we only ask to troubleshoot if `net_conntrack_dialer_conn_failed_total` says so.